kedro-org/kedro
# Make packaged Kedro project work in interactive environment

## Description
Part of https://github.com/kedro-org/kedro/issues/3237 and https://github.com/kedro-org/kedro/issues/1807 

Updated Databricks docs here: https://github.com/kedro-org/kedro/pull/4067

Follow up on https://github.com/kedro-org/kedro/pull/4012

>[!NOTE]
> If/when this PR is accepted, I'll update the starters. For the time being there's a PR with only the find_run_command refactoring changes: https://github.com/kedro-org/kedro-starters/pull/230


## Development notes
Similar changes have been proposed before, e.g. https://github.com/kedro-org/kedro/issues/2682 and tried in https://github.com/kedro-org/kedro/pull/3191. Note that there was a bit of muddled communication though around what value to set `standalone_mode` to. In many conversations it says setting `standalone_mode` to `True`, but that's the default and for an interactive environment like ipython or Databricks it should be set to `False`.

## How to test
1. Checkout branch and install branch.
2. Create a new project (double check it has the modified `__main__.py` from this PR)
3. Add some sample pipeline code to run.
4. `kedro package`
5. `pip install dist/<package>`
6. `ipython`
7. `import <package>`
8. `from <package>.__main__ import main()`
9. `main()`
10. Your project will run without error.

To compare this PR to the current status, follow the above steps with the latest released kedro and you'll see that when running `main()` inside ipython you'll get an `SystemExit: 0`.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist
